### PR TITLE
StringBuilder - name changes for the 'great renaming'

### PR DESCRIPTION
**Pull request by stephentetley**
Hi Magnus - here are the `StringBuilder` name changes.

This includes the new implementation of `StringBuilder.append!`.

I've also included `appendLineWith!` `appendLines!` and `appendLinesWith!` that have been written for the String Additions changes. I'm happy to remove them and resubmit them with the String changes if you just want a "minimum" pull request.

Array depends on StringBuilder so the `Array.flix` and `TestArray.flix` files have changed.

### CODE REVIEW

File: main/src/library/StringBuilder.flix [hunk 1]
```diff
@@ -113,15 +117,36 @@ namespace StringBuilder {
...
+    ///
+    /// Use the function `f` to converted the element `x` to a string with `f` append it, followed by the
+    /// system line separator, to the StringBuilder `sb`.
+    ///
+    pub def appendLineWith!(sb: StringBuilder, f: a -> String, x: a): Unit & Impure =
+        sb.appendString!(f(x));
+        sb.appendLineSeparator!()
+
+    ///
+    /// Append the each string in array `a` as a separate line to the StringBuilder `sb`.
```
Comment (magnus-madsen) [inline main/src/library/StringBuilder.flix:139]: How about:

Appends each string in the array `a` to the string builder `sb`.

File: main/src/library/StringBuilder.flix [hunk 2]
```diff
@@ -113,15 +117,36 @@ namespace StringBuilder {
...
-        sb.appendLineSeparator()
+    pub def appendLine!(sb: StringBuilder, s: String): Unit & Impure =
+        sb.appendString!(s);
+        sb.appendLineSeparator!()
+
+    ///
+    /// Use the function `f` to converted the element `x` to a string with `f` append it, followed by the
+    /// system line separator, to the StringBuilder `sb`.
+    ///
+    pub def appendLineWith!(sb: StringBuilder, f: a -> String, x: a): Unit & Impure =
```
Comment (magnus-madsen) [inline main/src/library/StringBuilder.flix:134]: How about:

Appends `f(x)` to the string builder `sb`.

or something like that?

File: main/src/library/StringBuilder.flix [hunk 3]
```diff
@@ -113,15 +117,36 @@ namespace StringBuilder {
...
+
+    ///
+    /// Append the each string in array `a` as a separate line to the StringBuilder `sb`.
+    ///
+    pub def appendLines!(sb: StringBuilder, a: Array[String]): Unit & Impure =
+        Array.foreach(appendLine!(sb), a)
+
+    ///
+    /// Append the each element in array `a` as a separate line to the StringBuilder `sb`, using `f`.
+    /// to convert each element of `a` to a String.
```
Comment (magnus-madsen) [inline main/src/library/StringBuilder.flix:146]: Update doc to be in sync with the two above.

**Comment by stephentetley** [comment]
Thanks Magnus - I've pushed a change to update both.

**Comment by magnus-madsen** [comment]
> 
> 
> Thanks Magnus - I've pushed a change to update both.

Can I also get an update to `appendLineWith` - thanks :)

**Comment by stephentetley** [comment]
> Can I also get an update to `appendLineWith` - thanks :)

Will do.

What about this for `appendLinesWith!`  - its a bit more indicative that we are processing an array?

~~~

    ///
    /// Appends `f(x)` for each x in array `a` to the string builder `sb`.
    ///
    pub def appendLinesWith!(sb: StringBuilder, f: a -> String, a: Array[a]): Unit & Impure =
        Array.foreach(appendLineWith!(sb, f), a)

~~~

**Comment by magnus-madsen** [comment]
Sounds goods!

**Comment by stephentetley** [comment]
I've pushed the revisions.

**Comment by magnus-madsen** [comment]
Merged. Thanks Stephen!

**Pull request merged**